cfme/web_ui/utilization.py

```diff
@@ -0,0 +1,790 @@
+        elif self.option.get_interval(force_visible_text=True) == Option.IN_HOURLY:
+            bucket_duration = "1h"
+            # sample date: 10/25/2016
+            t_dt = datetime.strptime(self.option.get_date(), '%m/%d/%Y')
+            start_dt = dt.replace(year=t_dt.year, month=t_dt.month, day=t_dt.day)
+            stop_dt = start_dt.replace(hour=23, minute=59, second=59)
+        elif self.option.get_interval(force_visible_text=True) == Option.IN_MOST_RECENT_HOUR:
+            bucket_duration = "1mn"
+            # sample date: 2016-10-26 07:15:48 UTC to 2016-10-26 07:30:48 UTC
+            date_range = self.option.get_range().split("to")
+            start_dt = datetime.strptime(date_range[0].strip(), "%Y-%m-%d %H:%M:%S %Z")
+            stop_dt = datetime.strptime(date_range[1].strip(), "%Y-%m-%d %H:%M:%S %Z")
+        start = self._timestamp(dt=start_dt)
+        end = self._timestamp(dt=stop_dt, end_time=True)
+        return self._list_data_mgmt(start=start, end=end, bucket_duration=bucket_duration)
+
+    def _list_data_mgmt(self, start, end, bucket_duration):
+        page = self._utz_object.page
+        mgmt = page.provider.mgmt
+        page_type = page.taggable_type
+        if page_type == "MiddlewareServer":
+            feed_id = page.feed
+            server_name = page.name
+        elif page_type == "MiddlewareDatasource":
+            feed_id = page.server.feed
+            server_name = page.server.name
+            resource_id = re.search(r'Datasource \[(.*?)\]', page.name).group(1)
+        elif page_type == "MiddlewareMessaging":
+            feed_id = page.server.feed
+            server_name = page.server.name
+            if page.name.startswith("JMS Topic"):
+                resource_id = re.search(r'JMS Topic \[(.*?)\]', page.name).group(1)
+                is_topic = True
+            elif page.name.startswith("JMS Queue"):
+                resource_id = re.search(r'JMS Queue \[(.*?)\]', page.name).group(1)
+                is_topic = False
+
+        # MiddlewareServer
+        if self.id == "jvm_heap_usage_bytes" and page_type == "MiddlewareServer":
+            raw_used = mgmt.metric.list_gauge_server(
+                feed_id=feed_id, server_id=server_name, bucket_duration=bucket_duration,
+                metric_enum=MetricEnumGauge.SVR_MEM_HEAP_USED, start=start, end=end)
+            raw_max = mgmt.metric.list_gauge_server(
+                feed_id=feed_id, server_id=server_name, bucket_duration=bucket_duration,
+                metric_enum=MetricEnumGauge.SVR_MEM_HEAP_MAX, start=start, end=end)
+            raw_committed = mgmt.metric.list_gauge_server(
+                feed_id=feed_id, server_id=server_name, bucket_duration=bucket_duration,
+                metric_enum=MetricEnumGauge.SVR_MEM_HEAP_COMMITTED, start=start, end=end)
+            # merge all data
+            final_data = self._update_mgmt_data(key='used', raw_data=raw_used)
```

Comment:
I don't see how this merges the data, and doesn't just overwrite final_data each time?